linkerd/http-route/src/grpc/filter/inject_failure.rs

```diff
@@ -10,3 +10,47 @@ pub struct FailureResponse {
     pub code: u16,
     pub message: std::sync::Arc<str>,
 }
+
+#[cfg(feature = "proto")]
+pub mod proto {
+    use super::*;
+    use linkerd2_proxy_api::grpc_route as api;
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum InvalidFailureResponse {
+        #[error("invalid gRPC status code: {0}")]
+        StatusNonU16(u32),
+
+        #[error("invalid request distribution: {0}")]
+        Distribution(#[from] rand::distributions::BernoulliError),
+    }
+
+    // === impl InjectFailure ===
+
+    impl TryFrom<api::GrpcFailureInjector> for InjectFailure {
+        type Error = InvalidFailureResponse;
+
+        fn try_from(proto: api::GrpcFailureInjector) -> Result<Self, Self::Error> {
+            let response = {
+                if proto.code > u16::MAX as u32 {
+                    return Err(InvalidFailureResponse::StatusNonU16(proto.code));
+                }
+
+                FailureResponse {
+                    code: proto.code as u16,
```

Comment:
it occurs to me that we _could_ use a checked cast here with `try_into`, rather than checking and then casting...but i'm not sure if that would actually make the code that much clearer, so this is fine.